samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Microsoft.AspNetCore.Components;
+using Cryptocash.Application.Dto;
+using Microsoft.SqlServer.Server;
+using Nox.Types;
+
+namespace Cryptocash.Ui.Generated.Component
+{
+#nullable enable
+
+    public class UiViewStreetAddressBase : ComponentBase
+    {
+        #region Declarations
+
+        [Parameter]
+        public StreetAddressDto? StreetAddress { get; set; }
+
+        [Parameter]
+        public List<CountryDto>? CountrySelectionList { get; set; }
+
+        #endregion
+
+         public string DisplayStreetAddress
```

Comment:
I would refactor this property in the following way:
1. Define a list with strings to be included to RtnStreetAddress -  `var parts = new List<string>();`
2. Create a function to add non empty strings to the list:
```
private void AddNonEmpty(string? value, List<string> parts)
{
    if (!string.IsNullOrWhiteSpace(value))
    {
        parts.Add(value.Trim());
    }
}
```
3. Replace all the checks to `AddNonEmpty(StreetAddress.StreetNumber, parts);`
4. Join the list with comma - `RtnStreetAddress = string.Join(",  ", parts).Trim(); `

Feel free to use other names for variables/functions, it's just an example.

Reply:
thanks @anna-naumova25  - that is a lot better - have made change in latest commit